go/enclave/db/sql/sql.go

```diff
@@ -19,18 +18,33 @@ const (
 	searchQry = `select * from keyvalue where substring(keyvalue.ky, 1, ?) = ? and keyvalue.ky >= ? order by keyvalue.ky asc`
 )
 
-// sqlEthDatabase implements ethdb.Database
-type sqlEthDatabase struct {
+// EnclaveSQLDB - exposes the underlying sql database
+type EnclaveSQLDB interface {
```

Comment:
It's hard to review the interfaces when they're not being used yet but I'd be tempted to get rid of both `EnclaveSQLDB` and `EnclaveDB`. Whenever I find myself writing `Impl` I think I'm probably thinking in a java-ey way haha.

If you just rename sqlEthDatabase to EnclaveDatabase or SQLEnclaveDB or something, since you're making SQL a requirement then there won't be alternative implementations. It can still implement ethdb.Database and provide the GetSQLDB() method. Wdyt?

Reply:
I'm not quite clear how they'll get used. Let me see in the next PR, and I'll get rid of them then if they serve no purpose